ts/features/bonus/bpd/components/bpdCardComponent/BpdCardComponent.tsx

```diff
@@ -0,0 +1,266 @@
+import { Badge, Text, View } from "native-base";
+import * as React from "react";
+import { Image, ImageBackground, StyleSheet } from "react-native";
+import { format } from "../../../../../utils/dates";
+import { H2 } from "../../../../../components/core/typography/H2";
+import { H5 } from "../../../../../components/core/typography/H5";
+import { IOColors } from "../../../../../components/core/variables/IOColors";
+import { BpdAmount } from "../../store/actions/amount";
+import { BpdPeriod } from "../../store/actions/periods";
+import { H4 } from "../../../../../components/core/typography/H4";
+import I18n from "../../../../../i18n";
+import bpdBonusLogo from "../../../../../../img/bonus/bpd/logo_BonusCashback_White.png";
+import bpdCardBgFull from "../../../../../../img/bonus/bpd/bonus_bg.png";
+import bpdCardBgPreview from "../../../../../../img/bonus/bpd/bonus_preview_bg.png";
+import { formatNumberCentsToAmount } from "../../../../../utils/stringBuilder";
+import IconFont from "../../../../../components/ui/IconFont";
+import TouchableDefaultOpacity from "../../../../../components/TouchableDefaultOpacity";
+
+type Props = {
+  period: BpdPeriod;
+  totalAmount: BpdAmount;
+  preview?: boolean;
+  onPress?: () => void;
+};
+
+const styles = StyleSheet.create({
+  flex1: {
+    flex: 1
+  },
+  flex2: {
+    flex: 2
+  },
+  container: {
+    flex: 1,
+    height: 192
+  },
+  paddedContentFull: {
+    paddingLeft: 16,
+    paddingTop: 24,
+    paddingRight: 20
+  },
+  paddedContentPreview: {
+    paddingLeft: 18,
+    paddingTop: 8,
+    paddingRight: 22
+  },
+  row: {
+    flexDirection: "row"
+  },
+  column: {
+    flexDirection: "column"
+  },
+  spaced: {
+    justifyContent: "space-between"
+  },
+  fullLogo: {
+    resizeMode: "contain",
+    height: 56,
+    width: 56,
+    alignSelf: "flex-end"
+  },
+  previewLogo: {
+    resizeMode: "contain",
+    height: 40,
+    width: 40,
+    alignSelf: "center"
+  },
+  badgeBase: {
+    alignSelf: "flex-end",
+    height: 18,
+    marginTop: 9
+  },
+  badgeTextBase: { fontSize: 12, lineHeight: 16 },
+  preview: {
+    marginBottom: -20,
+    height: 88
+  },
+  imageFull: {
+    resizeMode: "stretch",
+    height: 192
+  },
+  imagePreview: {
+    resizeMode: "stretch",
+    height: 88,
+    width: "100%"
+  },
+  amountTextBaseFull: {
+    fontSize: 24,
+    lineHeight: 35,
+    marginBottom: -8
+  },
+  amountTextUpperFull: { fontSize: 32 },
+  amountTextBasePreview: {
+    fontSize: 16,
+    lineHeight: 32
+  },
+  amountTextUpperPreview: { fontSize: 24 },
+  alignItemsCenter: {
+    alignItems: "center"
+  },
+  justifyContentCenter: {
+    justifyContent: "center"
+  }
+});
+
+type BadgeDefinition = {
+  style: any;
```

Comment:
```suggestion
  style: StyleProp<ViewStyle>;
```

Reply:
In order to avoid `any`

Reply:
I wanted to change it before opening the PR, but then missed it at the end. Thanks!